src/ensembl/src/header/headerActions.ts

```diff
@@ -1,11 +1,22 @@
-import { createStandardAction } from 'typesafe-actions';
+import { createAction, createStandardAction } from 'typesafe-actions';
+import analyticsTracking from 'src/services/analytics-service';
+import { CustomDimensions } from 'src/analyticsHelper';
 
 export const toggleAccount = createStandardAction('header/toggle-account')();
 
 export const toggleLaunchbar = createStandardAction(
   'header/toggle-launchbar'
 )();
 
-export const changeCurrentApp = createStandardAction(
-  'header/change-current-app'
-)<string>();
+export const changeCurrentApp = createAction(
+  'header/change-current-app',
+  (resolve) => {
+    return (appName: string) => {
+      analyticsTracking.setCustomDimension({
+        diemension: CustomDimensions.PAGEVIEW,
+        value: appName
+      });
+      return resolve(appName);
+    };
```

Comment:
to middleware?

Reply:
Created separate setter functions for dimensions as discussed.